app/controllers/addresses_controller.rb

```diff
@@ -1,21 +1,47 @@
 # frozen_string_literal: true
 
-# rubocop:disable Metrics/MethodLength
 class AddressesController < ApplicationController
   def update
     service = Service.find_by_id(params[:service_id])
     address = Address.find_by_id(params[:id])
     if service && address
-      if service.addresses.include?(address)
-        render status: :ok
-      else
-        service.addresses << address
-        service.save!
-        render status: :created
-      end
+      add_address_to_service(service, address)
+      return
     else
       render status: :bad_request
     end
   end
+
+  def destroy
+    service = Service.find_by_id(params[:service_id])
+    address = Address.find_by_id(params[:id])
+    if service && address
+      remove_address_from_service(service, address)
+      return
+    else
+      render status: :bad_request
+    end
+  end
+
+  private
+
+  def add_address_to_service(service, address)
+    if service.addresses.include?(address)
+      render status: :ok
+    else
+      service.addresses << address
+      service.save!
+      render status: :created
+    end
+  end
+
+  def remove_address_from_service(service, address)
+    if service.addresses.include?(address)
+      service.addresses.delete(address)
+      service.save!
```

Comment:
Note that it's probably not necessary ta call save here, since services and addresses are a many-to-many relationship, which means that deleting the relationship corresponds to deleting a row in the join table, but without actually changing either model.